EasyDisenchant.lua

```diff
@@ -14,9 +14,43 @@ local _M = {
 	buttonRenderingCache = {},
 };
 
+EasyDisenchantBlacklist = {};
+
 BINDING_HEADER_EASY_DISENCHANT = _M.addonName;
 BINDING_NAME_EASY_DISENCHANT_OPEN = SHOW;
 
+local IsBlacklisted = function(itemLink)
+    for i, BLitemLink in pairs(EasyDisenchantBlacklist) do
+        if(BLitemLink == itemLink) then
+            return true;
+        end
+    end
+    return false;
+end
+
+local Blacklist = function(itemLink)
+    if(EasyDisenchantBlacklist == {}) then
```

Comment:
This is bad. Every time the function `Blacklist` is called, you're creating an empty table and comparing it to `EasyDisenchantBlacklist`. Lua compares tables by reference, which means that this will never be true because they are two different tables (regardless of their content).